dataloader.py

```diff
@@ -142,7 +149,36 @@ def _one_hot_encode(self, image, mask):
         
         return image, one_hot_map
 
-    def data_batch(self, batch_size, augment, shuffle=False, one_hot_encode=False):
+    def _map_function(self, images_path, masks_path):
+        image, mask = self._parse_data(images_path, masks_path)
+
+        if self.augment:
+            if self.compose:
+                image, mask = self._corrupt_brightness(image, mask)
+                image, mask = self._corrupt_contrast(image, mask)
+                image, mask = self._corrupt_saturation(image, mask)
+                image, mask = self._crop_random(image, mask)
+                image, mask = self._flip_left_right(image, mask)
+            else:
+                options = [self._corrupt_brightness,
+                           self._corrupt_contrast,
+                           self._corrupt_saturation,
+                           self._crop_random,
+                           self._flip_left_right]
+                augment_func = random.choice(options)
+                image, mask = augment_func(images_path, masks_path)
```

Comment:
should be augment_func(image, mask) as we can't augment the path

Reply:
please check the last commit I have changed it.